arrow-optics/src/main/kotlin/arrow/optics/std/list.kt

```diff
@@ -1,25 +1,56 @@
 package arrow.optics
 
+import arrow.Kind
+import arrow.core.Either
 import arrow.core.ListExtensions
+import arrow.core.ListK
+import arrow.core.NonEmptyList
 import arrow.core.None
 import arrow.core.Option
 import arrow.core.Some
+import arrow.core.Tuple2
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.fix
 import arrow.core.identity
-import arrow.core.ListK
-import arrow.core.NonEmptyList
 import arrow.core.k
+import arrow.core.left
+import arrow.core.right
+import arrow.core.toOption
+import arrow.core.toT
+import arrow.optics.typeclasses.Cons
+import arrow.optics.typeclasses.FilterIndex
+import arrow.optics.typeclasses.Index
+import arrow.optics.typeclasses.Snoc
+import arrow.typeclasses.Applicative
+import kotlin.reflect.KClass
 
 /**
  * [Optional] to safely operate on the head of a list
  */
+@Deprecated(
+  "ListK is being deprecated, use the function defined for List instead.",
+  ReplaceWith("List::class.head<A>()"),
+  DeprecationLevel.WARNING)
```

Comment:
@raulraja @nomisRev What about this function? Should we keep `List` class as the receiver? Or should we use the `Optional` companion to project the function as `Optional.listHead<A>()`

Reply:
This are issues we'll only encounter in Optics afaik and it's unrelated to the typeclasses. We have the same issue in all other types, for example [`TupleN`](https://github.com/arrow-kt/arrow-optics/blob/master/arrow-optics/src/main/kotlin/arrow/optics/std/tuple.kt#L17) since these are just top-level APIs for Optics. Similar to `emptyList<A>()` and `listOf(1, 2, 3)` etc in Kotlin.

So we can either follow the Kotlin Std pattern of putting this simply as top-level functions, or we follow Paco's suggestion of `Optional.listHead<A>()` which I like better to be honest.

Reply:
Optional companion option sounds good 👍